src/main/java/org/zalando/planb/provider/AuthorizeController.java

```diff
@@ -1,42 +1,92 @@
 package org.zalando.planb.provider;
 
-import com.google.common.collect.ImmutableSet;
-import com.nimbusds.jose.JOSEException;
+import static java.lang.String.format;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import static org.zalando.planb.provider.OIDCController.getRealmName;
+import static org.zalando.planb.provider.OIDCController.validateRedirectUri;
+import static org.zalando.planb.provider.realms.ClientRealmAuthenticationException.clientNotFound;
+
+import java.io.IOException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
 import org.apache.http.client.utils.URIBuilder;
+
 import org.slf4j.Logger;
+
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.config.Scope;
+
+import org.springframework.http.MediaType;
+
 import org.springframework.stereotype.Controller;
+
 import org.springframework.ui.Model;
+
 import org.springframework.web.bind.annotation.RequestHeader;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.RedirectView;
+
 import org.zalando.planb.provider.realms.ClientRealm;
 import org.zalando.planb.provider.realms.UserRealm;
 import org.zalando.planb.provider.realms.UserRealmAuthenticationException;
 
-import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
+import com.google.common.collect.ImmutableSet;
 
-import static java.lang.String.format;
-import static org.slf4j.LoggerFactory.getLogger;
-import static org.zalando.planb.provider.realms.ClientRealmAuthenticationException.clientNotFound;
-import static org.zalando.planb.provider.OIDCController.getRealmName;
-import static org.zalando.planb.provider.OIDCController.validateRedirectUri;
+import com.nimbusds.jose.JOSEException;
 
 @Controller
+@RequestMapping(path = AuthorizeController.AUTHORIZE_ENDPOINT)
 public class AuthorizeController {
 
-    static final Set<String> SUPPORTED_RESPONSE_TYPES = ImmutableSet.of("code", "token");
```

Comment:
I already started refactoring and putting them in various classes. 
I can't find a perfect model for this... perfect would be finding a library full with constants and use them, i guess.

Schema could be *.contstants.Parameres. bla bla or *.constants.Headers.Field1 etc etc